src/durableorchestrationclient.ts

```diff
@@ -46,8 +47,14 @@ const URL = url.URL;
  * };
  * ```
  */
-export function getClient(context: unknown): DurableOrchestrationClient {
-    let clientData = getClientData(context as IOrchestrationFunctionContext);
+export function getClient(
+    context: IOrchestrationFunctionContext,
+    clientInputOptions: OrchestrationClientInput

```

Comment:
What is this new `clientInputOptions` parameter?

Reply:
Unlike the orchestrator, activity, and entity functions, client functions don't have a special trigger. Instead, they are normal triggered functions and the client comes from an extra input binding. In the old model, the `getClient` function would just loop over all the available bindings on the `Context` object and find the one that matches the `durableClient` input binding. The new programming model doesn't work like that anymore. Instead, to register an input binding, this is how you would register and access it:

```TS
const clientInput = input.generic({
    type: 'orchestrationClient'
});

app.generic('DurableFunctionsHttpStart', {
    trigger: trigger.http(),
    extraInputs: [clientInput],
    handler: async function (context: InvocationContext, req: HttpRequest): Promise<any> {
        const clientInputBinding = context.getInput(clientInput)
    }
})
```

So essentially, the `getClient` method would need to have access to the `FunctionInput` object that the customer added to their `extraInputs` property, so that it can call `context.getInput()` on behalf of the customer. The type `OrchestrationClientInput` is just a `FunctionInput` object with the `type` property set to `durableClient`.

Regardless, this shouldn't matter because all of this should be abstracted away from the customer, and we would provide an API that allows them to register a client function that just gets passed the client directly as its third parameter, like this:

```TS
df.httpClient('DurableFunctionsHttpStart', async (context: InvocationContext, req: HttpRequest, client: DurableOrchestrationClient) => {
    const instanceId = await client.startNew(req.query.get('functionName'), undefined, req.body);
    context.log(`Started orchestration with ID = '${instanceId}'.`);
    return client.createCheckStatusResponse(req, instanceId);
})
```

This is just a step to get there